### PR TITLE
RainbowWallet Refactor

### PR DESCRIPTION
**Pull request by crypblizz8**
[Loom Demo]()
Click here for [demo video](https://www.loom.com/share/a06971027a2e4e6796a2e53a63189c1d)

**Context**
Hacked on the privy-quick starter guide during the ETHCC Hack. 
Used Rainbowkit during my hack and refactored this starter guide with it. 

Why: Having a great Developer Experiencer (DX) is crucial and making it inviting as much as possible. 

### Approach / Main files worked on:
- ```App.js```
    - Added the Wagmi + Rainbowkit client as a wrapper
- ```Index.js```
    - No need to update Address as this is tracked by Wagmi library
    - Moved the rendering components into smaller components. 
- ```Globals.css```
    - Added Header Div 
    - Removed the body font. 

### In order to test
- Add in privy API key
- Ensure you have the three fields in your console. Follow the guide here.[]()
- ```yarn```
- ```yarn dev```

###  User Flow Edge Cases to consider / test
**First-time User**
- Connect to Rainbowkit
- Auth with the module
- Add in your user Data
- Button should say “Pushed to Privy”
- Check Privy DB
- Refresh and see if data is updated. 

**Existing User (Connected Wallet)**
- Page should load with all previous data
- Input new Data
- Refresh and see if data is updated. 

**Existing User (Not connected wallet)**
- Connect to Wallet
- Old Data should appear. 
- Input new Data
- Refresh and see if data is updated. 

**Existing User / Disconnect Wallet.** 
- Disconnect Wallet
- Page should reset
- Connect Wallet Again
- Old Data should appear. 
- Input new Data
- Refresh and see if data is updated.

### Next Steps
- Test the several user flows / edge cases 
- Provide feedback 
- Adjust the code accordingly
- Remove the unnecessary comments.

### CODE REVIEW

**Comment by nichochar** [comment]
@crypblizz I like the contribution. I have some minor feedback on the code but I wanted to give a higher level review first:
I actually think it would be great to have a WAGMI + RainbowKit quickstart, since many people are familiar with those libraries, however one of the design goals of this quickstart was to be as lean as possible, in order to teach about the Privy libraries.
To a newcomer, it could be not obvious what is wagmi, what is rainbowkit and what is privy, but to a user familiar with them, your solution is better.
Therefore my suggestion is to actually create a whole new repo from your branch in the code, and name it privy-quickstart-wagmi-rainbowkit. 

I will review the code so we can clean it up, and once we're happy with it, instead of merging it, we will boostrap it to a new repo.

File: pages/index.js [hunk 1]
```diff
@@ -13,67 +15,104 @@ const client = new PrivyClient({
 });
 
 export default function Home() {
-  // Use React's useState hook to keep track of the signed in Ethereum address.
+  // Use Wagmi Address instead of settingState through privy client.
```
Comment (nichochar) [inline pages/index.js:18]: Let's improve this comment by not mentioning `settingState through privy Client`. Simply:
```
// Use WAGMI provider for logging in and getting the eth Address.
```

File: pages/index.js [hunk 2]
```diff
@@ -13,67 +15,104 @@ const client = new PrivyClient({
...
+  const { address: ethAddress, isConnected, isDisconnected } = useAccount();
+
+  // Use React's useState hook to keep track of the signed in Ethereum address
   const [address, setAddress] = useState(null);
```
Comment (nichochar) [inline pages/index.js:22]: Do we still need this? Could we simply not use the `address` directly set on line 19, and not use `setAddress` on line 34

File: pages/index.js [hunk 3]
```diff
@@ -13,67 +15,104 @@ const client = new PrivyClient({
...
+  const [authenticatedwithPrivy, setAuthenticatedwithPrivy] = useState(false);
+  const [savedData, setSavedData] = useState(false);
 
-      if (!ethereum) {
-        alert("Please install MetaMask for this demo: https://metamask.io/");
-        return;
-      }
+  // Moved all useEffects to top for readability.
+  // Initial Auth.
+  useEffect(() => {
```
Comment (nichochar) [inline pages/index.js:33]: See comment above. We can likely remove this `useEffect` entirely and get the address directly from `useAccount`

File: pages/index.js [hunk 4]
```diff
@@ -13,67 +15,104 @@ const client = new PrivyClient({
...
+    document.body.style = `background: ${favoriteColor};`;
+    if (isDisconnected) {
+      setFirstName("");
+      setDateOfBirth("");
+      setFavoriteColor("");
+      document.body.style = `background: white;`;
+    }
+  }, [favoriteColor, isDisconnected]);
+
+  // Auth pattern already asks to sign when you putUserData / client.put()
```
Comment (nichochar) [inline pages/index.js:57]: Let's remove this code if it's not being used

File: pages/index.js [hunk 5]
```diff
@@ -13,67 +15,104 @@ const client = new PrivyClient({
...
+      ]);
+      setFirstName(name.text());
+      setDateOfBirth(birthday.text());
+      setFavoriteColor(color.text());
+      if (name && birthday && color) {
+        console.log("Successfully wrote user data to Privy.");
+        setSavedData(true);
+      }
+    } catch (error) {
+      console.log("Pushing Privy Data error.", error);
```
Comment (nichochar) [inline pages/index.js:97]: Nit, can we change the error message to `"Error when writing data to Privy:", error`

File: pages/index.js [hunk 6]
```diff
@@ -13,67 +15,104 @@ const client = new PrivyClient({
...
 
-  // Update address if page is refreshed.
-  const updateAddress = async () => {
-    const address = await session.address();
-    setAddress(address);
-  };
-  useEffect(() => {
-    updateAddress();
-  }, []);
+  // Commented out due to Rainbowkit already tracking...
```
Comment (nichochar) [inline pages/index.js:101]: Let's remove rather than comment out

File: pages/index.js [hunk 7]
```diff
@@ -87,68 +126,87 @@ export default function Home() {
...
+      </div>
+    </div>
+  );
 
-  // Set background to user's favorite color.
-  useEffect(() => {
-    if (!favoriteColor) return;
-    document.body.style = `background: ${favoriteColor};`;
-  }, [favoriteColor]);
+  // If the original connect Wallet button is necessary, can add back in.
```
Comment (nichochar) [inline pages/index.js:140]: Same as above, let's remove

File: styles/globals.css [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 body {
   text-align: center;
-  font-size: 1.8rem;
+  /* font-size: 1.8rem; */
```
Comment (nichochar) [inline styles/globals.css:5]: also remove

File: styles/globals.css [hunk 9]
```diff
@@ -74,3 +74,10 @@ button:disabled {
 button:active {
   transform: scale(0.99);
 }
+
+/* Header with RainbowWallet */
```
Comment (nichochar) [inline styles/globals.css:78]: No need for this comment, the className is very descriptive already!
